finality-aleph/src/sync/ticker.rs

```diff
@@ -0,0 +1,143 @@
+        sleep(self.current_periodic_timeout.saturating_sub(since_last)).await;
+        self.current_periodic_timeout = self.max_timeout;
+        self.last_broadcast = Instant::now();
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use tokio::time::{sleep, timeout, Duration};
+
+    use super::BroadcastTicker;
+
+    #[tokio::test]
+    async fn try_broadcast() {
+        let max_timeout = Duration::from_millis(700);
+        let min_timeout = Duration::from_millis(100);
+        let mut ticker = BroadcastTicker::new(max_timeout, min_timeout);
+
+        assert!(!ticker.try_broadcast());
+        sleep(min_timeout).await;
+        assert!(ticker.try_broadcast());
+        assert!(!ticker.try_broadcast());
+    }
+
+    #[tokio::test]
+    async fn wait_for_periodic_broadcast() {
+        let max_timeout = Duration::from_millis(700);
+        let min_timeout = Duration::from_millis(100);
+        let mut ticker = BroadcastTicker::new(max_timeout, min_timeout);
+
+        assert_ne!(
+            timeout(2 * min_timeout, ticker.wait_for_periodic_broadcast()).await,
+            Ok(())
+        );
+        assert_eq!(
+            timeout(max_timeout, ticker.wait_for_periodic_broadcast()).await,
+            Ok(())
+        );
+    }
+
+    #[tokio::test]
+    async fn wait_for_periodic_broadcast_after_try_broadcast_true() {
+        let max_timeout = Duration::from_millis(700);
+        let min_timeout = Duration::from_millis(100);
+        let mut ticker = BroadcastTicker::new(max_timeout, min_timeout);
+
+        sleep(min_timeout).await;
+        assert!(ticker.try_broadcast());
+
+        assert_ne!(
+            timeout(2 * min_timeout, ticker.wait_for_periodic_broadcast()).await,
+            Ok(())
+        );
+        assert_eq!(
+            timeout(max_timeout, ticker.wait_for_periodic_broadcast()).await,
+            Ok(())
+        );
+    }
+
+    #[tokio::test]
+    async fn wait_for_periodic_broadcast_after_try_broadcast_false() {
+        let max_timeout = Duration::from_millis(700);
+        let min_timeout = Duration::from_millis(100);
+        let mut ticker = BroadcastTicker::new(max_timeout, min_timeout);
+
+        assert!(!ticker.try_broadcast());
+
+        assert_eq!(
+            timeout(2 * min_timeout, ticker.wait_for_periodic_broadcast()).await,
+            Ok(())
+        );
+        assert_ne!(
+            timeout(2 * min_timeout, ticker.wait_for_periodic_broadcast()).await,
+            Ok(())
+        );
+        assert_eq!(
+            timeout(max_timeout, ticker.wait_for_periodic_broadcast()).await,
+            Ok(())
+        );
+    }
+
+    #[tokio::test]
+    async fn try_broadcast_after_wait_for_periodic_broadcast() {
+        let max_timeout = Duration::from_millis(700);
+        let min_timeout = Duration::from_millis(100);
+        let mut ticker = BroadcastTicker::new(max_timeout, min_timeout);
+
+        assert_eq!(
+            timeout(
+                max_timeout + min_timeout,
```

Comment:
Why not `2 * max_timeout` like everywhere else?

Reply:
No need to wait so much, but I think I will create two constants. One a little bit over `MIN_TIMEOUT` and one a little bit over `MAX_TIMEOUT`